### PR TITLE
[backend/frontend] Improv team performance

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Proposed changes

* Change teams call to use new teams endpoint to improv performance
* Change component when select teams on create and update inject (in scenario/simulation/atomic testing)

### CODE REVIEW

**Comment by savacano28** [comment]
Hi! Is it possible to avoid using fetchTeams in injectUpdateDetails? The get /api/teams endpoint uses a query that needs to be refactored.
useDataLoader(() => {
    dispatch(fetchTeams());
    dispatch(fetchTags());
  });

**Comment by savacano28** [comment]
I don't think dispatch(fetchTeams()) is used.

**Comment by savacano28** [comment]
I think this: dispatch(fetchTeams());, it is not used

File: openbas-front/src/admin/components/common/injects/InjectAddTeams.tsx [hunk 1]
```diff
@@ -43,97 +34,91 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 interface Props {
   handleAddTeams: (teamIds: string[]) => void;
-  injectTeamsIds: string[]
-  teams: TeamStore[]
+  injectTeamsIds: string[];
 }
```
Comment (savacano28) [inline openbas-front/src/admin/components/common/injects/InjectAddTeams.tsx:38]: QuickInject has still teams param

File: openbas-front/src/admin/components/scenarios/scenario/teams/ScenarioTeams.tsx [hunk 2]
```diff
@@ -65,8 +65,8 @@ export const teamContextForScenario = (scenarioId: ScenarioStore['scenario_id'],
         dispatch(enableScenarioTeamPlayers(scenarioId, teamId, { scenario_team_players: [userId] }));
       }
     },
-    searchTeams(input: SearchPaginationInput): Promise<{ data: Page<TeamOutput> }> {
-      return searchScenarioTeams(scenarioId, input);
+    searchTeams(input: SearchPaginationInput, contextualOnly?: boolean): Promise<{ data: Page<TeamOutput> }> {
+      return searchScenarioTeams(scenarioId, input, contextualOnly);
     },
```
Comment (savacano28) [inline openbas-front/src/admin/components/scenarios/scenario/teams/ScenarioTeams.tsx:70]: Maybe it's possible to replace fetchTeams (lines 37 and 41) with searchScenarioTeams as well?

File: openbas-front/src/admin/components/simulations/simulation/teams/ExerciseTeams.tsx [hunk 3]
```diff
@@ -59,8 +59,8 @@ export const teamContextForExercise = (exerciseId: ExerciseStore['exercise_id'],
         dispatch(enableExerciseTeamPlayers(exerciseId, teamId, { exercise_team_players: [userId] }));
       }
     },
-    searchTeams(input: SearchPaginationInput): Promise<{ data: Page<TeamOutput> }> {
-      return searchExerciseTeams(exerciseId, input);
+    searchTeams(input: SearchPaginationInput, contextualOnly?: boolean): Promise<{ data: Page<TeamOutput> }> {
+      return searchExerciseTeams(exerciseId, input, contextualOnly);
     },
```
Comment (savacano28) [inline openbas-front/src/admin/components/simulations/simulation/teams/ExerciseTeams.tsx:64]: Maybe it's possible to replace fetchTeams (lines 31 and 37) with searchScenarioTeams as well?

File: openbas-front/src/admin/components/common/injects/UpdateInjectDetails.js [hunk 4]
```diff
@@ -355,7 +354,6 @@ const UpdateInjectDetails = ({
                       handleClose={handleClose}
                       tagsMap={tagsMap}
                       permissions={permissions}
-                      teamsFromExerciseOrScenario={teamsFromExerciseOrScenario}
                       articlesFromExerciseOrScenario={[]}
```
Comment (savacano28) [inline openbas-front/src/admin/components/common/injects/UpdateInjectDetails.js:357]: Maybe it's possible to replace/remove  fetchTeams (lines 73) ?

File: openbas-front/src/admin/components/common/injects/UpdateInject.tsx [hunk 5]
```diff
@@ -81,7 +79,6 @@ const UpdateInject: React.FC<Props> = ({ open, handleClose, onUpdateInject, mass
             handleClose={handleClose}
             onUpdateInject={onUpdateInject}
             isAtomic={isAtomic}
-            teamsFromExerciseOrScenario={teamsFromExerciseOrScenario}
             {...props}
```
Comment (savacano28) [inline openbas-front/src/admin/components/common/injects/UpdateInject.tsx:82]: When I try to update an atomicTesting, I have any teams on his definition: 
image

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingUpdate.tsx [hunk 6]
```diff
@@ -63,7 +46,6 @@ const AtomicTestingUpdate: FunctionComponent<Props> = ({
       onUpdateInject={onUpdateAtomicTesting}
       injectId={atomic.inject_id}
       isAtomic
-      teamsFromExerciseOrScenario={teams?.filter((team: TeamStore) => !team.team_contextual) ?? []}
     />
```
Comment (savacano28) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingUpdate.tsx:49]: I think here we lost the teams of atomic testing

Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingUpdate.tsx:49]: Fixed

File: openbas-front/src/admin/components/common/injects/UpdateInjectDetails.js (hunk 4 above)
Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/common/injects/UpdateInjectDetails.js:355]: Remote it

File: openbas-front/src/admin/components/scenarios/scenario/teams/ScenarioTeams.tsx (hunk 2 above)
Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/scenarios/scenario/teams/ScenarioTeams.tsx:70]: Should be done in an other PR

**Review by savacano28 — APPROVED** [review]

**Pull request merged**